gcloud-java-compute/src/main/java/com/google/gcloud/compute/RegionAddressId.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Identity for a Google Compute Engine region address.
+ */
+public final class RegionAddressId extends RegionResourceId implements AddressId {
```

Comment:
Ok. I was more thinking of having `abstract AddressId` extended by both `RegionAddressId` and `GlobalAddressId` which would conceptually make sense. It is true that these classes are not too long, and if it consistent with something else, then no problem. I just wanted to engage that thought.